src/definitions/modules/progress.js

```diff
@@ -394,19 +419,54 @@ $.fn.progress = function(parameters) {
         },
 
         set: {
-          barWidth: function(value) {
-            if(value > 100) {
-              module.error(error.tooHigh, value);
-            }
-            else if (value < 0) {
-              module.error(error.tooLow, value);
+          barWidth: function(values) {
+            module.debug("set bar width with ", values);
+            values = toArray(values);
+            var total = values.reduce(function(a,b) {
+              return a + b;
+            }, 0);
+            if(total > 100) {
+              module.error(error.tooHigh, total);
+            }
+            else if (total < 0) {
+              module.error(error.tooLow, total);
             }
             else {
-              $bar
-                .css('width', value + '%')
-              ;
+              var firstNonZeroIndex = -1;
+              var lastNonZeroIndex = -1;
+              var percents = values.map(function(value, index) {
+                var $bar = $($bars[index]);
+                if (value === 0) {
+                  $bar.css('display', 'none');
+                } else {
+                  if (firstNonZeroIndex == -1) {
+                    firstNonZeroIndex = index;
+                  }
+                  lastNonZeroIndex = index;
+                  $bar.css('display', 'block');
+                  $bar.css('width', value + '%');
+                }
+                return parseInt(value, 10);
+              });
+              values.forEach(function(_, index) {
+                var $bar = $($bars[index]);
+                if (index == firstNonZeroIndex) {
```

Comment:
Could be merged as
```javascript
$bar.css({
    borderTopLeftRadius: index==firstNonZeroIndex ? '' : 0,
    borderBottomLeftRadius: index==firstNonZeroIndex ? '' : 0,
    borderTopRightRadius: index==lastNonZeroIndex ? '' : 0,
    borderBottomRightRadius: index==lastNonZeroIndex ? '' : 0
})
```

Reply:
Done in https://github.com/fomantic/Fomantic-UI/pull/582/commits/43c64890618f1ee873b764fb1dbb6911fc0ddd76